tasks/src/main/java/ru/sberbank/school/task02/calculator/ExtendedCurrencyCalculator.java

```diff
@@ -0,0 +1,108 @@
+        List<Quote> finalQuoteList = new ArrayList<>();
+        BigDecimal countCurr, volumeInCurr;
+        for (Quote quote: quotes) {
+            if (operation == ClientOperation.SELL) {
+                countCurr = quote.getBid();
+            } else {
+                countCurr = quote.getOffer();
+            }
+            volumeInCurr = quote.getVolumeSize().multiply(countCurr).setScale(10, RoundingMode.HALF_UP);
+            if (quote.getVolume().isInfinity() || volumeInCurr.compareTo(amountOfRequest) > 0) {
+                finalQuoteList.add(quote);
+            }
+        }
+        if (finalQuoteList.size() > 0) {
+            finalQuoteList.sort(comparator);
+            System.out.println(currentDate.toString() + " Sorted list: ");
+            showQuotes(finalQuoteList);
+            return Optional.of(finalQuoteList.get(0));
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public Optional<BigDecimal> convertReversed(ClientOperation operation, Symbol symbol, BigDecimal amount, Beneficiary beneficiary) {
+        quotes = externalQuotesService.getQuotes(Symbol.USD_RUB);
+
+        if (operation == null) {
+            throw new NullPointerException("operation is null!");
+        }
+        if (beneficiary == null) {
+            throw new NullPointerException("beneficiary is null!");
+        }
+
+        this.symbolOfRequest = symbol;
+        this.amountOfRequest = amount;
+        System.out.println("Get benificiary: " + beneficiary.toString());
+        System.out.println("Get symbol: " + symbol.getSymbol());
+        System.out.println("Get client operation: " + operation.toString());
+        System.out.println("Get request volume: " +  amountOfRequest );
+        showQuotes(quotes);
+//        if ((beneficiary == Beneficiary.BANK && operation == ClientOperation.BUY)
+//                || (beneficiary == Beneficiary.CLIENT && operation == ClientOperation.SELL)) {
+//            rounding_mode = RoundingMode.HALF_UP;
+//        }
+//        if ((beneficiary == Beneficiary.BANK && operation == ClientOperation.SELL)
+//                || (beneficiary == Beneficiary.CLIENT && operation == ClientOperation.BUY)) {
+//            rounding_mode = BigDecimal.HALF_UP;
+//        }
+        this.amountOfRequest = amount.setScale(10,  rounding_mode);
+
+        if (check()) {
+            return extendedOperation(operation, beneficiary, quotes);
+        }
+        System.out.println("Return Optional.empty() after check");
+        return Optional.empty();
+    }
+
+    @Override
+    public Optional<BigDecimal> convertReversed(ClientOperation operation,
+                                                Symbol symbol,
+                                                BigDecimal amount,
+                                                double delta,
+                                                Beneficiary beneficiary) {
+         return Optional.empty();
```

Comment:
&#39;method def&#39; child has incorrect indentation level 9, expected level should be 8.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-1163007983>*